File: 0169-majority-element/0169-majority-element.java
public class Solution {
    public int majorityElement(int[] nums) {
        Arrays.sort(nums);
        int lastNum = Integer.MAX_VALUE;
        int lastNumFreq = 0;
        for (int num : nums) {
            if (lastNum != num) {
                lastNum = num;
                lastNumFreq = 1;
            } else {
                lastNumFreq++;
            }
            if (lastNumFreq > nums.length / 2) {
                return lastNum;
            }
        }
        return 0;
    }
}

File: 0169-majority-element/0169-majority-element.swift
class Solution {
    func majorityElement(_ nums: [Int]) -> Int {
        var counter = [Int: Int]()
        for num in nums {
            counter[num, default: 0] += 1
            if counter[num]! > nums.count / 2 {
                return num
            }
        }
        return 0
    }
}